codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -281,11 +313,16 @@ class SymbolVisitor(
     override fun memberShape(shape: MemberShape): Symbol {
         val target = model.expectShape(shape.target)
         val targetSymbol = this.toSymbol(target)
+        if (config.handleOptionality && config.handleRequired) {
```

Comment:
If we end up using two flags, I'd check this once upon `SymbolVisitorConfig` creation and never again.